### ISSUE TITLE
Streamed uploads

### ISSUE THREAD

**Issue by mpol** [httpie/httpie#201]
It would be sometimes very nice to be able to PUT a large file without buffering it first.
And it seems that requests supports the chunked encoding well: http://docs.python-requests.org/en/latest/user/advanced/#chunk-encoded-requests

Creating a generator in `_body_from_file` if a `--chunked` option is set seems simple enough, but I guess something would have to be done for `--print=B` to work (or maybe suppress the output with a reasonable message - since the (or at least my) use case would be large, generally binary, files).

**Comment by sigmavirus24** [comment]
Stupid question: is there a way to do `multipart/form-data` uploads with httpie? If so, you'll want to stream those eventually too and there's a bunch of utilities in the [requests-toolbelt](/sigmavirus24/requests-toolbelt) to help with this including a way to monitor the upload of a streamed `multipart/form-data` request.

**Comment by jakubroztocil** [comment]
@sigmavirus24 yep, [you can upload files as well](https://github.com/jakubroztocil/httpie#file-upload-forms):

```bash
$ http -f POST example.com/jobs name='John Smith' cv@~/Documents/cv.pdf
```
I'll take a look at the features provided by requests-toolbelt. Thanks for the tip!

**Comment by totalvoidness** [comment]
Just to be sure, as I don't want to create a duplicate ticket: When I try to upload big files (2GB), I get the following error:

    http: error: ConnectionError: ('Connection aborted.', error(22, 'Invalid argument')

I think this is related to this issue, as the file can not yet be uploaded chunked?

**Comment by jkbrzt** [comment]
Upstream issues: 

* [ ] kennethreitz/requests#3035
* [ ] kevin1024/pytest-httpbin#33

**Comment by jakubroztocil** [comment]
Related #684 #685

**Comment by jakubroztocil** [comment]
Streamed uploads are implemented in `master`. It will be included in v2.3.0.
